app/models/whitehall_migration/asset_import.rb

```diff
@@ -25,9 +25,10 @@ def content_publisher_asset
   end
 
   def whitehall_asset_id
-    url_array = original_asset_url.to_s.split("/")
     # https://github.com/alphagov/asset-manager#create-an-asset
-    url_array[url_array.length - 2]
+    path = original_asset_url.sub(/.+?(?=\/government)/, "")
```

Comment:
Just a suggestion, [URI](https://apidock.com/ruby/URI) has you covered...

```
original_asset_url = 'https://static.gov.uk/government/uploads/system/uploads/image_data/file/847150/foo.jpg'
=> "https://static.gov.uk/government/uploads/system/uploads/image_data/file/847150/foo.jpg"

path = original_asset_url.sub(/.+?(?=\/government)/, "")
=> "/government/uploads/system/uploads/image_data/file/847150/foo.jpg"

path = URI(original_asset_url).path           
=> "/government/uploads/system/uploads/image_data/file/847150/foo.jpg"
```

Reply:
Good shout Graham 👍 I originally thought it would be important to look for a `/government` prefix, but actually Asset Manager will return an error if it can't find the given path anyway, so this can definitely be simplified to your suggestion.